Dockerfile.horizon

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19 AS builder
+FROM golang:1.20.1 AS builder
```

Comment:
I think we want to leave the patch version off here. That way every time this is run, people would be pulling in the latest golang image, so they would get security/etc patch updates as they are available. This is less deterministic but I think it makes more sense to have this match the behavior of the the [package builder](https://github.com/stellar/pipelines/pull/528); ie you get what you get at build time